### PR TITLE
Nexus 22100 conan remove component problem

### PR DESCRIPTION
**Pull request by igorudovika**
NEXUS-22100 Fixed problem when asset deletion procedure lefts wrong component

### CODE REVIEW

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ConanComponentMaintenanceImpl.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      return super.deleteAssetTx(assetId, deleteBlob);
+    }
+
+    final Component component = tx.findComponent(componentId);
+    if (component == null) {
+      return Collections.emptySet();
+    }
+    return deleteComponentTx(componentId, deleteBlob).getAssets();
+  }
+}
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ConanComponentMaintenanceImpl.java]: new line

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ConanComponentMaintenanceImpl.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.transaction.TransactionalDeleteBlob;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import static org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.State.STARTED;
+
+/**
+ * @since 0.0.next
+ */
+public class ConanComponentMaintenanceImpl
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ConanComponentMaintenanceImpl.java]: It seems to me you've got single relationship (single component has single asset). Why don't you use `SingleAssetComponentMaintenance`?

Comment (igorudovika) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ConanComponentMaintenanceImpl.java]: Yep, make sense: 165ddce2a9b1f386ab1864f70181664c59a166d5

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ConanComponentMaintenanceImpl.java (hunk 1 above)
Comment (igorudovika) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ConanComponentMaintenanceImpl.java]: deleted this file 165ddce2a9b1f386ab1864f70181664c59a166d5

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/rest/ConanProxyRepositoriesApiResource.java [hunk 3]
```diff
@@ -15,21 +15,37 @@
...
+import io.swagger.annotations.ApiResponses;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+
+import static org.sonatype.nexus.rest.ApiDocConstants.AUTHENTICATION_REQUIRED;
+import static org.sonatype.nexus.rest.ApiDocConstants.INSUFFICIENT_PERMISSIONS;
+import static org.sonatype.nexus.rest.ApiDocConstants.REPOSITORY_CREATED;
+import static org.sonatype.nexus.rest.ApiDocConstants.REPOSITORY_NOT_FOUND;
+import static org.sonatype.nexus.rest.ApiDocConstants.REPOSITORY_UPDATED;
+
 @Named
```
Comment (mlukaretkyi) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/rest/ConanProxyRepositoriesApiResource.java]: you forgot @Api(value = API_REPOSITORY_MANAGEMENT)

Comment (igorudovika) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/rest/ConanProxyRepositoriesApiResource.java]: Fixed 5ec733ed3293c46b3126af127fc30a040e05a962

**Pull request merged**